packages/flutter_tools/lib/src/android/deferred_components_setup_validator.dart

```diff
@@ -0,0 +1,766 @@
+    }
+    encodedMapping = encodedMapping.substring(0, encodedMapping.length - 1); // remove trailing comma.
+    // Check for existing metadata entry and see if needs changes.
+    bool exists = false;
+    bool needsChange = false;
+    for (final XmlElement metaData in document.findAllElements('meta-data')) {
+      final String name = metaData.getAttribute('android:name');
+      if (name == _mappingKey) {
+        exists = true;
+        final String storedMappingString = metaData.getAttribute('android:value');
+        if (storedMappingString != encodedMapping) {
+          needsChange = true;
+        }
+      }
+    }
+    if (needsChange || !exists) {
+      lines = appManifestFile.readAsLinesSync();
+      List<String> cleanedLines = <String>[];
+      if (exists) {
+        // Remove existing mapping entry. We do this line by line
+        // to maintain the formatting and comments in the file.
+        for (int i = 0; i < lines.length; i++) {
+          final String line = lines[i];
+          if (line.trim().startsWith(_metadataEntryMatcher)) {
+            // Reconstruct metadata entry regardless of how many lines it is split over
+            String metaDataLine = '';
+            int j = i;
+            while (!lines[j].contains('/>')) {
+              metaDataLine += lines[j];
+              j++;
+            }
+            metaDataLine += lines[j];
+            if (metaDataLine.contains(_mappingMetadataMatcher)) {
+              // found the relevant to remove. Skip adding to outLines.
+              i = j; // the for loop will increment this one more time next loop.
+            } else {
+              cleanedLines.add(line);
+            }
+          } else {
+            cleanedLines.add(line);
+          }
+        }
+      } else {
+        cleanedLines = lines;
+      }
+      // Create new metadata entry.
+      final File manifestOutput = _outputDir
+        .childDirectory('app')
+        .childDirectory('src')
+        .childDirectory('main')
+        .childFile('AndroidManifest.xml');
+      if (manifestOutput.existsSync()) {
+        manifestOutput.deleteSync();
+      }
+      manifestOutput.createSync(recursive: true);
+      for (final String line in cleanedLines) {
+        if (line.contains(_manifestApplicationEntryCloseMatcher)) {
+          final String indent = line.substring(0, line.indexOf('</application>'));
+          final String additionalIndent = indent.length > 2 ? '    ' : indent;
+          manifestOutput.writeAsStringSync(
+              '$indent$additionalIndent<meta-data android:name="$_mappingKey" android:value="$encodedMapping" />\n',
+              mode: FileMode.append, flush: true);
+        }
+        manifestOutput.writeAsStringSync('$line\n', mode: FileMode.append, flush: true);
+      }
+      _modifiedFiles.add(manifestOutput.path);
+    }
+  }
+
+  /// Checks if the base module `app`'s `strings.xml` contain
+  /// string resources for each component's name.
```

Comment:
Could you expand on this a little more, what string resources are required, what is the expected format, et cetera